streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopicManagerTest.java

```diff
@@ -782,16 +782,18 @@ public void shouldCompleteTopicValidationOnRetry() {
         // let the first describe succeed on topic, and fail on topic2, and then let creation throws topics-existed;
         // it should retry with just topic2 and then let it succeed
         when(admin.describeTopics(Set.of(topic1, topic2)))
-            .thenAnswer(answer -> new MockDescribeTopicsResult(mkMap(
-                mkEntry(topic1, topicDescriptionSuccessFuture),
-                mkEntry(topic2, topicDescriptionFailFuture)
-            )));
+                .thenAnswer(answer -> new MockDescribeTopicsResult(mkMap(
+                        mkEntry(topic1, topicDescriptionSuccessFuture),
+                        mkEntry(topic2, topicDescriptionFailFuture) // first call: missing
+                )))
+                .thenAnswer(answer -> new MockDescribeTopicsResult(mkMap(
+                        mkEntry(topic1, topicDescriptionSuccessFuture),
+                        mkEntry(topic2, topicDescriptionSuccessFuture) // second call: now exists
+                )));
         when(admin.createTopics(Collections.singleton(new NewTopic(topic2, Optional.of(1), Optional.of((short) 1))
             .configs(mkMap(mkEntry(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT),
                                  mkEntry(TopicConfig.MESSAGE_TIMESTAMP_TYPE_CONFIG, "CreateTime"))))))
             .thenAnswer(answer -> new MockCreateTopicsResult(Collections.singletonMap(topic2, topicCreationFuture)));
-        when(admin.describeTopics(Collections.singleton(topic2)))
-            .thenAnswer(answer -> new MockDescribeTopicsResult(Collections.singletonMap(topic2, topicDescriptionSuccessFuture)));
```

Comment:
We should remove the code change from above, and keep this one, because on retry, we should only query for `topic2`.